File: frontend/src/api/teachingTask.ts
import request from '../utils/request'
import type { ApiResponse, PageResult } from './types'

/** 周次类型：ALL 全周 / ODD 单周 / EVEN 双周（与后端 WeekType 对齐，V9 单双周支持） */
export type WeekType = 'ALL' | 'ODD' | 'EVEN'

export interface TeachingTask {
  id: number
  courseId: number
  courseName: string
  teacherId: number
  teacherName: string
  classId: number
  className: string
  weeklyHours: number
  weekType: WeekType
  /** 连续周段起始周（闭区间，默认1，V10 连续周段支持） */
  startWeek: number
  /** 连续周段结束周（闭区间，默认20，V10 连续周段支持） */
  endWeek: number
  needContinuous: number
  status: number
  remark: string
  scheduledSlots: number
  requiredSlots: number
  createTime: string
  updateTime: string
}

export interface TeachingTaskForm {
  courseId: number
  teacherId: number
  classId: number
  weeklyHours: number
  weekType: WeekType
  /** 连续周段起始周（闭区间，默认1，V10 连续周段支持） */
  startWeek?: number
  /** 连续周段结束周（闭区间，默认20，V10 连续周段支持） */
  endWeek?: number
  needContinuous?: number
  status?: number
  remark?: string
}

export function getTeachingTaskList(params: {
  courseName?: string
  teacherName?: string
  className?: string
  status?: number
  semesterId?: number
  pageNum?: number
  pageSize?: number
}) {
  return request.get<ApiResponse<PageResult<TeachingTask>>>('/teaching-tasks', { params }).then((r) => {
    if (!r.data) {
      throw new Error('响应数据为空')
    }
    return r.data.data
  })
}

export function getTeachingTaskById(id: number) {
  return request.get<ApiResponse<TeachingTask>>(`/teaching-tasks/${id}`).then((r) => {
    if (!r.data) {
      throw new Error('响应数据为空')
    }
    return r.data.data
  })
}

export function createTeachingTask(data: TeachingTaskForm) {
  return request.post<ApiResponse<TeachingTask>>('/teaching-tasks', data).then((r) => {
    if (!r.data) {
      throw new Error('响应数据为空')
    }
    return r.data.data
  })
}

export function updateTeachingTask(id: number, data: TeachingTaskForm) {
  return request.put<ApiResponse<TeachingTask>>(`/teaching-tasks/${id}`, data).then((r) => {
    if (!r.data) {
      throw new Error('响应数据为空')
    }
    return r.data.data
  })
}

export function deleteTeachingTask(id: number) {
  return request.delete(`/teaching-tasks/${id}`)
}

export function getAllTeachingTasks() {
  return request.get<ApiResponse<TeachingTask[]>>('/teaching-tasks/all').then((r) => {
    if (!r.data) {
      throw new Error('响应数据为空')
    }
    return r.data.data
  })
}

File: backend/src/main/java/com/paike/scheduler/entity/TeachingTask.java
package com.paike.scheduler.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;

@Getter
@Setter
@TableName("teaching_task")
public class TeachingTask {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long semesterId;

    private Long courseId;

    private Long teacherId;

    private Long classId;

    private Integer weeklyHours;

    /** 周次类型：ALL全周、ODD单周、EVEN双周（V9 单双周支持） */
    private String weekType;

    /** 连续周段起始周（闭区间，默认1，V10 连续周段支持） */
    private Integer startWeek;

    /** 连续周段结束周（闭区间，默认20，V10 连续周段支持） */
    private Integer endWeek;

    private Integer needContinuous;

    private Integer status;

    private String remark;

    @TableLogic
    private Integer deleted;

    @TableField("create_time")
    private LocalDateTime createTime;

    @TableField("update_time")
    private LocalDateTime updateTime;
}
